### PR TITLE
Image Import Wrapper resource regionalization

### PR DESCRIPTION
**Pull request by zoran15**
- Creating scratch bucket in the same region as source file. If bucket already exists, use it without creating it.
- Using a zone from within source file's region (if source file is in multi-regional bucket, Cloud build is used)
- Various refactorings

Note: files in /mocks folder are auto generated to support gomock in unit tests. Please don't review them.

### CODE REVIEW

File: cli_tools/gce_vm_image_import/util/storage_client.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+func (hc *HTTPClient) Get(url string) (resp *http.Response, err error) {
+	return hc.httpClient.Get(url)
+}
+
+// NewStorageClient creates a StorageClient
+func NewStorageClient(ctx context.Context, client *storage.Client) *StorageClient {
+	o := []option.ClientOption{option.WithScopes(storagev1.DevstorageReadOnlyScope)}
+	hc, _, err := htransport.NewClient(ctx, o...)
+	if err != nil {
+		log.Fatalf("Cannot create storage HTTP client %v", err.Error())
```
Comment (adjackura) [inline cli_tools/gce_vm_image_import/util/storage_client.go]: Don't use fatal logging outside of main, instead return the error and handle it in the callers.

File: cli_tools/gce_vm_image_import/util/scratch_bucket_creator.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+		return "", "", fmt.Errorf("can't create scratch bucket if project not specified")
+	}
+
+	if sourceFileFlag != "" {
+		// source file provided, create bucket in the same region for cost/performance reasons
+		bucket, region, err = c.createBucketMatchFileRegion(sourceFileFlag, project)
+	}
+
+	if sourceFileFlag == "" || err != nil {
+		// source file not provided or couldn't create bucket based on it, create default scratch bucket
```
Comment (adjackura) [inline cli_tools/gce_vm_image_import/util/scratch_bucket_creator.go]: This woudl be more idiomatic:

if sourceFileFlag == "" || err != nil {
  bucket := c.formatScratchBucketName(project, defaultRegion)
  region, err = c.createBucketIfNotExisting(bucket, project, &storage.BucketAttrs{Name: bucket, Location: defaultRegion, StorageClass: defaultStorageClass})
  if err != nil {
    return "", "", err
  }
}
return bucket, region, nil

File: cli_tools/gce_vm_image_import/util/compute_service.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+import "google.golang.org/api/compute/v1"
+
+// ComputeService wraps GCE compute.Service and implements ComputeServiceInterface
+type ComputeService struct {
+	Cs *compute.Service
+}
+
+// GetZones returns a slice of Zones for give project
+func (cs *ComputeService) GetZones(project string) ([]*compute.Zone, error) {
```
[the remark this answers is not in the document]
Comment (zoran15) [reply cli_tools/gce_vm_image_import/util/compute_service.go]: Switched ComputeService to use compute.Client for ListZones. Kept ComputeService for mocking purposes.

File: cli_tools/gce_vm_image_import/util/storage_client.go (hunk 1 above)
Comment (zoran15) [reply cli_tools/gce_vm_image_import/util/storage_client.go]: Done

File: cli_tools/gce_vm_image_import/util/scratch_bucket_creator.go (hunk 2 above)
Comment (zoran15) [reply cli_tools/gce_vm_image_import/util/scratch_bucket_creator.go]: Done

**Pull request merged**